packages/api/src/controllers/v1/agit/mod.rs

```diff
@@ -1,95 +1,197 @@
+        extract::{Path, Query, State},
+        routing::{get, post},
+    },
+};
+use by_types::QueryResponse;
+use models::{
+    Result,
+    error::ServiceError as ApiError,
+    v1::agit::{
+        Agit, AgitAction, AgitByIdAction, AgitCreateRequest, AgitGetResponse, AgitParam, AgitQuery,
+        AgitRepository, AgitSummary, AgitUpdateRequest,
+    },
+};
+use sqlx::postgres::PgRow;
 
 #[derive(Clone, Debug)]
 pub struct AgitControllerV1 {
-    pool: sqlx::PgPool,
     repo: AgitRepository,
+    pool: sqlx::Pool<sqlx::Postgres>,
 }
 
 impl AgitControllerV1 {
-    pub fn new(pool: sqlx::PgPool) -> Self {
-        let repo = Agit::get_repository(pool.clone());
-        Self { repo, pool }
+    async fn query(
+        &self,
+        auth: Option<Authorization>,
+        param: AgitQuery,
+    ) -> Result<QueryResponse<AgitSummary>> {
+        tracing::debug!("{param}");
+        if auth.is_none() {
+            return Err(ApiError::Unauthorized);
+        }
+
+        let mut total_count = 0;
+        let items: Vec<AgitSummary> = AgitSummary::query_builder()
+            .limit(param.size())
+            .page(param.page())
+            .query()
+            .map(|row: PgRow| {
+                use sqlx::Row;
+                total_count = row.try_get("total_count").unwrap_or_default();
+                row.into()
+            })
+            .fetch_all(&self.pool)
+            .await?;
+
+        Ok(QueryResponse { total_count, items })
     }
 
-    pub fn route(pool: sqlx::PgPool) -> Result<by_axum::axum::Router> {
-        let ctrl = Self::new(pool.clone());
-        Ok(by_axum::axum::Router::new()
-            .route("/:id", post(Self::act_by_id).get(Self::get))
-            .route("/", post(Self::act).get(Self::list))
-            .with_state(ctrl)
-            .nest(
-                "/:agit_id/collections",
-                CollectionControllerV1::route(pool.clone())?,
-            ))
+    async fn create(
+        &self,
+        auth: Option<Authorization>,
+        AgitCreateRequest { title }: AgitCreateRequest,
+    ) -> Result<Agit> {
+        if auth.is_none() {
+            return Err(ApiError::Unauthorized);
+        }
+
+        let agit = self.repo.insert(title).await?;
+
+        Ok(agit)
+    }
+
+    async fn update(
+        &self,
+        id: i64,
+        auth: Option<Authorization>,
+        param: AgitUpdateRequest,
+    ) -> Result<Agit> {
+        if auth.is_none() {
+            return Err(ApiError::Unauthorized);
+        }
+
+        Agit::query_builder()
+            .id_equals(id)
+            .query()
+            .fetch_one(&self.pool)
+            .await
+            .map_err(|_| ApiError::NotFound)?;
+
+        let agit = self.repo.update(id, param.into()).await?;
+        Ok(agit)
+    }
+
+    async fn delete(&self, id: i64, auth: Option<Authorization>) -> Result<Agit> {
+        if auth.is_none() {
+            return Err(ApiError::Unauthorized);
+        }
+
+        Agit::query_builder()
```

Comment:
Why does it need to get before delete? I think the delete cause error if no data, right?